Make delegation methods of `std::net::IpAddr` unstably const

CDirkx opened the pull request:
Make the following methods of `std::net::IpAddr` unstable const under the `const_ip` feature:
 - `is_unspecified`
 - `is_loopback`
 - `is_global`
 - `is_multicast`

Also adds a test for these methods in a const context.

Possible because these methods delegate to the inner `Ipv4Addr` or `Ipv6Addr`, which were made const ([PR#76205](https://github.com/rust-lang/rust/pull/76142) and [PR#76206](https://github.com/rust-lang/rust/pull/76206)), and the recent stabilization of const control flow.

Part of #76205

r? @ecstatic-morse

ecstatic-morse commented:
LGTM! Any reason not to do `is_ipv{4,6}` as part of this PR as well?

@bors delegate+

bors commented:
:v: @CDirkx can now approve this pull request
<!-- homu: {"type":"Delegated","delegator":"ecstatic-morse","delegate":"CDirkx"} -->

CDirkx commented:
I had already opened up [PR#76226](https://github.com/rust-lang/rust/pull/76226) to directly stabilize those as const, although I could include them here as unstable const, and when this gets merged rebase the stabilization PR.

ecstatic-morse commented:
No need.

@bors r+ rollup

bors commented:
:pushpin: Commit 0c339b5bf283bfc839c06018190b2ab849be9270 has been approved by `ecstatic-morse`

<!-- @bors r=ecstatic-morse 0c339b5bf283bfc839c06018190b2ab849be9270 -->
<!-- homu: {"type":"Approved","sha":"0c339b5bf283bfc839c06018190b2ab849be9270","approver":"ecstatic-morse"} -->

bors commented:
:umbrella: The latest upstream changes (presumably #76422) made this pull request unmergeable. Please resolve the merge conflicts.

CDirkx committed:
Make delegation methods of `std::net::IpAddr` unstable const

Make the following methods of `std::net::IpAddr` unstable const under the `const_ip` feature:
- `is_unspecified`
- `is_loopback`
- `is_global`
- `is_multicast`

Also adds a test for these methods in a const context.

Possible because these methods delegate to the inner `Ipv4Addr` or `Ipv6Addr`, which were made const, and the recent stabilization of const control flow.

Part of #76205

CDirkx commented:
Rebased onto master.

CDirkx commented:
@bors retry

bors commented:
:umbrella: The latest upstream changes (presumably #76850) made this pull request unmergeable. Please resolve the merge conflicts.

*Note that reviewers usually do not review pull requests until merge conflicts are resolved!* Once you resolve the conflicts, you should change the labels applied by bors to indicate that your PR is ready for review. Post this as a comment to change the labels:
```
@rustbot modify labels: +S-waiting-on-review -S-waiting-on-author
```

ecstatic-morse commented:
Huh, this seems to have never gotten added to the queue. `bors retry` can sometimes be finicky.

CDirkx commented:
Ah, I thought I used the wrong command. Rebased, so @bors retry

ecstatic-morse commented:
@bors r+

bors commented:
:pushpin: Commit 947536fca06e6dc6c3a3dd4ff1f66fc5080e9cbb has been approved by `ecstatic-morse`

<!-- @bors r=ecstatic-morse 947536fca06e6dc6c3a3dd4ff1f66fc5080e9cbb -->
<!-- homu: {"type":"Approved","sha":"947536fca06e6dc6c3a3dd4ff1f66fc5080e9cbb","approver":"ecstatic-morse"} -->

ecstatic-morse commented:
(I would use r={reviewer} when delegated to, retry seems to fail sometimes)
